### PR TITLE
Windoor fixes and tweaks

### LINKED ISSUE #939: Windoors are not affected by power loss
When an area loses power, any windoors in that area don't change behavior in any way:
- They still require ID access to open
- AI/borgs can still remotely open/close the windoors
- They cannot be toggled open/closed with a crowbar (Although for a windoor, I'd just make it openable without ID with an empty hand when there's no power)

### LINKED ISSUE #4951: It's possible to place items on a table, if there's a windoor blocking the table and it's closed
If you walk up to a table with a windoor preventing you from picking things up on the table, you're able to actually place items through the windoor, onto the table itself. This has been confirmed at the HoP Office, and the RnD lab. It's likely able to be done anywhere a windoor is blocking a table.

### LINKED ISSUE #6151: Tables Have Several Exploits When Behind Windoors - Items Placing On Them - Can Be Dismantled
Title would be took long if I listed everything.
- You can place objects through windoors in some areas, allowing you to bypass the windoor to do so.
- You can dismantle some tables behind their windoors, bypassing the windoor, as well. 
Neither require the door to be open and both can be done even if you don't have the access to do so.

### LINKED ISSUE #6229: More Windoor bugs
You can shank, heal, and drug anyone directly behind a windoor. You cannot use your bear hands however.

### PR DESCRIPTION
**Pull request by JohnWildkins**
Fixes #939, fixes #4951, fixes #6151, and fixes #6229.

Windoors are now affected by power. When power is lost in an area, non-reinforced windoors "fail open" - that is to say they remain shut, but can be pushed open with relative ease and don't have access checks, much like their airlock cousins. Reinforced windoors (aka security/brigdoors) fail closed and lock into place. Neither can be remotely accessed by borgs/AI while unpowered.

Probably also fixes some unrelated bugs about being able to attack/interact with things through objects you shouldn't normally.

### CODE REVIEW

**Comment by JohnWildkins** [comment]
!review

File: code/game/machinery/doors/windowdoor.dm [hunk 1]
```diff
@@ -130,7 +136,8 @@
 		return
 
 /obj/machinery/door/window/attack_ai(mob/user as mob)
-	return src.attack_hand(user)
+	if(operable())
```
Comment (Amunak) [inline code/game/machinery/doors/windowdoor.dm:139]: Maybe just remove the check? You already check it in attack_hand.

File: code/game/machinery/doors/windowdoor.dm [hunk 2]
```diff
@@ -215,6 +227,16 @@
...
 
+/obj/machinery/door/window/brigdoor/allowed(mob/M)
+	if(inoperable()) // Brigdoors are the exception to the "fail open" windoor - they lock closed
+		return FALSE
+	. = ..()
+
+/obj/machinery/door/window/brigdoor/power_change()
+	..()
+	if((stat & NOPOWER) && !density)
+		close(TRUE)
```
Comment (Amunak) [inline code/game/machinery/doors/windowdoor.dm:238]: Maybe add a message (perhaps in limited range) like "as the power shuts down the windoor closes".

File: code/game/machinery/doors/windowdoor.dm [hunk 3]
```diff
@@ -215,6 +227,16 @@
 	health = 300.0 //Stronger doors for prison (regular window door health is 150)
 
 
+/obj/machinery/door/window/brigdoor/allowed(mob/M)
+	if(inoperable()) // Brigdoors are the exception to the "fail open" windoor - they lock closed
+		return FALSE
+	. = ..()
```
Comment (Amunak) [inline code/game/machinery/doors/windowdoor.dm:233]: Perhaps this should generate a message of some kind to tell people why it doesn't open?

File: code/game/machinery/doors/windowdoor.dm [hunk 4]
```diff
@@ -142,7 +149,9 @@
 			user.visible_message("<span class='danger'>[user] smashes against [src].</span>", "<span class='danger'>You smash against [src]!</span>")
```
Comment (Amunak) [inline code/game/machinery/doors/windowdoor.dm:149]: The check could probably just be

```
var/mob/living/carbon/human/H = user
if(H && H.species.can_shred(H))
```

**Review by Amunak — APPROVED** [review]

File: code/game/machinery/doors/windowdoor.dm (hunk 2 above)
Comment (JohnWildkins) [reply code/game/machinery/doors/windowdoor.dm:238]: Eh, that message doesn't exist for airlocks opening/closing on power loss and in places like the brig it'd just spam the chat log.

File: code/game/machinery/doors/windowdoor.dm (hunk 3 above)
Comment (JohnWildkins) [reply code/game/machinery/doors/windowdoor.dm:233]: Yeah, not a bad idea.

File: code/game/machinery/doors/windowdoor.dm (hunk 2 above)
Comment (Amunak) [reply code/game/machinery/doors/windowdoor.dm:232]: True. I wonder if showing a message (for _all_ doors that do this) in at least a very small range (maybe 2/3 tiles) would be meaningful, but I guess power outage is a visible enough even that it doesn't make much sense. Though I personally was quite surprised when I saw doors open without power, lol.

**Review by Amunak — APPROVED** [review]

**Review by mikomyazaki — APPROVED** [review]

**Pull request merged**